File: src/go/calculatepi/calculatepi.go
package calculatepi

func CalculatePi(terms []int) []float64 {
  var denominator float64
  var operation float64
	pi := make([]float64, len(terms))

	for i := 0; i < len(terms); i++ {
		denominator = 1.0
		operation = 1.0
		for cont := 0; cont < terms[i]; cont++ {
			pi[i] += operation * (4 / denominator)
			denominator += 2
			operation *= -1
		}
	}
	return pi
}

File: src/scala/CalculatePi.scala

def calculatePi(n: Int): Double = {
  (0 until n).foldLeft(0.0)((pi, index) => {
    val operation: Double = if (index % 2 == 0) 1.0 else -1.0
    val denominator: Double = 1.0 + (index * 2.0)
    pi + (operation * (4.0 / denominator))
  })
}

object Main extends App {
  val nTerms: Seq[Int] = Seq(10, 1000, 100000, 10000000)
  nTerms.foreach(n => println(s"Pi ($n): ${calculatePi(n)}"))
}
